src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -1232,17 +1232,20 @@ internal enum ErrorCode
         ERR_FixedBufferTooManyDimensions = 7092,
         ERR_CantReadConfigFile = 7093,
         ERR_BadAwaitInCatchFilter = 7094,
-        WRN_FilterIsConstant = 7095,
+        // WRN_FilterIsConstant = 7095,   // replaced by FilterIsConstantFalse and FilterIsConstantTrue
         ERR_EncNoPIAReference = 7096,
         //ERR_EncNoDynamicOperation = 7097,   // dynamic operations are now allowed
         ERR_LinkedNetmoduleMetadataMustProvideFullPEImage = 7098,
         ERR_MetadataReferencesNotSupported = 7099,
         ERR_InvalidAssemblyCulture = 7100,
         ERR_EncReferenceToAddedMember = 7101,
         ERR_MutuallyExclusiveOptions = 7102,
+        WRN_FilterIsConstantFalse = 7103,
+        WRN_FilterIsConstantTrue = 7104,
+        WRN_FilterIsConstantRedundantTryCatch = 7105,
```

Comment:
The answer to that question would set the direction of this pull request (may need to close and have a new one to remove the filter altogether?). I doubt this kind of code will be written in a practical application, but still seems like a nice hint if #19294 is to be merged as it skips redundant catch clause emission if unreachable. 
Also, I am trying to think if any sort of automatic refactoring done by IDE could potentially write a dummy boolean value in the filter expression. In that case, if the developer forgets to replace the dummy value, it would at least come up as a warning in compilation.